messaging/src/main/java/org/axonframework/serialization/upcasting/event/EventTypeUpcaster.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2010-2020. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.serialization.upcasting.event;
+
+import org.axonframework.serialization.SerializedType;
+import org.axonframework.serialization.SimpleSerializedType;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * A {@link SingleEventUpcaster} implementation which allows for type upcasting only. This could be used if the event's
+ * class name did not follow the desired naming convention or if an event's package name has been adjusted.
+ * <p>
+ * Note that this upcaster <b>should not</b> be used to change the semantic meaning of an event. Such a requirement
+ * points towards a new event type instead of adjusting an existing one.
+ *
+ * @author Steven van Beelen
+ * @since 4.3
+ */
+public abstract class EventTypeUpcaster extends SingleEventUpcaster {
+
+    /**
+     * Retrieve the expected event payload type this upcaster should react on.
+     *
+     * @return the expected event payload type this upcaster should react on
+     */
+    public abstract String expectedPayloadType();
```

Comment:
Curious, why an abstract class and not a class where the expectedPayloadType, expectedRevision and their upcast countertypes are passed as constructor parameters?

Reply:
That's, a good question. Guess I felt like making an abstract class with abstract methods?
Regardless, I feel a constructor is more sensible too, as it'll require less work from our users. Hence, I'll change this around.